### ISSUE TITLE
LT 4.3 increases memory of LO with every change of text

### ISSUE THREAD

**Issue by FredKruse** [languagetool-org/languagetool#1144]
The used memory of LO grows with every change of text (also small once). A test with 4.2 shows that this is only a problem of 4.3.
The used heap size does not grow at all. It varies in the expected way. The analysis of the introduced cache shows no grows (The number of paragraphs is constant as long as no new paragraph is added).
I'm a little bit helpless because I don't understand why the used memory of LO grows without a enlargement of the java heap size.

**Comment by danielnaber** [comment]
Thanks, but any idea *why* this solves the memory issue?

**Comment by FredKruse** [comment]
No, I fear, I don't understand the java memory management. I did two things:
1. Now, the FlatParagraph object is used only locally inside of the SingleDocument class.
2. The MessageHandler is changed to a static class with static variables.
Before the change the in crease of memory was dramatically. The LO memory blows up by working on a large document till LO crashed because it tried to use more memory as available. I compared the new code with the code of 4.2 and did a lot of try and error.
Now it works without increasing the memory (it uses is a little bit more than 4.2. This I expected because of using the cash). But I don't understand really why.
By the way I'm working on a linux system and openjdk. I don't tested LT on a windows or other system. Maybe this is a problem of the virtual machine.
